pkg/database-types/mysql.go

```diff
@@ -54,20 +57,36 @@ func (m *mysql) getCredentials(vaultClientConfiguration map[interface{}]interfac
 
 func (m *mysql) TranslateFybrikConfigToOpenMetadataConfig(config map[string]interface{},
 	connectionType string, credentials *string) map[string]interface{} {
+	ret := make(map[string]interface{})
 	if m.vaultClientConfiguration != nil && credentials != nil {
 		username, password, err := m.getCredentials(m.vaultClientConfiguration, credentials)
 		if err == nil && username != EmptyString && password != EmptyString {
-			config[Username] = username
-			config[Password] = password
+			ret[Username] = username
+			ret[Password] = password
 		}
 	}
-	return config
+	host, ok1 := config[Host]
+	port, ok2 := config[Port]
+	if ok1 && ok2 {
+		ret[HostPort] = fmt.Sprintf("%s:%.0f", host, port)
```

Comment:
In `TranslateFybrikConfigToOpenMetadataConfig`, the `config` parameter is of type `map[string]interface{}`.
For some reason, the port value, which appears as `"port": 3306` in the JSON request, has type `float64` in `config`.
That is why I use `fmt.Sprintf("%s:%.0f", host, port)`.
`Atoi(s string)` is not appropriate since the `port` is not represented as a string.
Regarding your second question, I think that it is okay to ignore the case that the `host` is present but the `port` is missing. Both values are required fields in the taxonomy.

Reply:
Can it be a default port `3306` ?